tests/system_tests/test_blueapi_system.py

```diff
@@ -58,6 +59,22 @@ def client_with_stomp() -> BlueapiClient:
     )
 
 
+@pytest.fixture(scope="module", autouse=True)
+def wait_for_server():
+    client = BlueapiClient.from_config(config=ApplicationConfig())
+
+    attempts_remaining = 20
+    while attempts_remaining > 0:
+        try:
+            client.get_environment()
+            return
+        except ConnectionError:
+            ...
+        attempts_remaining -= 1
+        time.sleep(0.5)
+    raise TimeoutError("No connection to the blueapi server")
```

Comment:
```suggestion
    for _ in range(20):
        try:
            client.get_environment()
            return
        except ConnectionError:
            ...
        time.sleep(0.5)
    raise TimeoutError("No connection to the blueapi server")
```

nit: Why keep track when Python'll do it for ya mate

Reply:
Good shout, see also https://github.com/DiamondLightSource/blueapi/issues/763